File: 90-subsets-ii/subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        res = []
        def myFun(index,arr):
            res.append(arr[:])
            for j in range(index,len(nums)):
                if j > index and nums[j] == nums[j-1]:
                    continue
                arr.append(nums[j])
                myFun(j+1,arr)
                arr.pop(-1)
        myFun(0,[])
        return res

File: 90-subsets-ii/subsets-ii.java
class Solution {
    public List<List<Integer>> result;
    public int[] numsG;
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        result = new ArrayList<>();
        numsG = nums;
        Arrays.sort(numsG);
        fun(0,new ArrayList<>());
        return result;
    }
    public void fun(int index, List<Integer> curr){
        if(index >= numsG.length){
            result.add(new ArrayList<>(curr));
            return;
        }
        curr.add(numsG[index]);
        fun(index + 1, curr);
        curr.remove(curr.size()-1);
        while(index < numsG.length-1 && numsG[index] == numsG[index + 1]){
            index++;
        }
        fun(index + 1, curr);
    }
}